### PR TITLE
DataGenerator readability improvements

### PR DESCRIPTION
**Pull request by gregschohn**
### Description

Rewrite the workload classes to use a fluent style w/ mixins so that the structure of what's being cons'ed together is much more immediately clear.  Case in point, making the conversion took hours because 1) there weren't unit tests on the generator itself and 2) the one test that was there didn't have suitable logging - requests weren't printed out (which is probably why there what looked to be a bug in the docIds).

This also maintains the previous status-quo of the codebase in how to eschew static imports and shows how to favor mixins and for DSL-like (DSL-lite) applications.

* Category Refactoring
* Why these changes are required? First, to find out why there was pushback to [removing static imports](https://github.com/opensearch-project/opensearch-migrations/pull/1076#discussion_r1808796774).  Then I realized that I couldn't understand the code & decided that if this was going to be used more often, we needed it to be clearer and to have better patterns in place.
* What is the old behavior before changes and new behavior after changes? docIds no longer contain a space.

### Issues Resolved
Same sonar cleanup as [this PR](https://github.com/opensearch-project/opensearch-migrations/pull/1076), which was merged before this one was opened for review.

### Testing
gradle testing

### Check List
- [ ] New functionality includes testing
  - [x] All tests pass, including unit test, integration test and doctest
- [ ] New functionality has been documented
- [x] Commits are signed per the DCO using --signoff

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

File: DataGenerator/src/test/resources/log4j2.properties [hunk 1]
```diff
@@ -6,7 +6,7 @@ appender.console.type = Console
 appender.console.name = Console
 appender.console.target = SYSTEM_OUT
 appender.console.layout.type = PatternLayout
-appender.console.layout.pattern = %m%n
+appender.console.layout.pattern = %d{yyyy-MM-dd HH:mm:ss,SSS}{UTC} %p %c{1.} [%t] %m%n
```
Comment (gregschohn) [inline DataGenerator/src/test/resources/log4j2.properties:9]: Keeping thread identifiers is critical for multi-threaded code like this.  It's necessary to correlate log lines together.

File: DataGenerator/src/main/java/org/opensearch/migrations/data/WorkloadGenerator.java [hunk 2]
```diff
@@ -43,11 +45,25 @@ public void generate(WorkloadOptions options) {
...
         // This happens inline to be sure the index exists before docs are indexed on it
-        client.createIndex(indexName, workload.createIndex(options.index.indexSettings.deepCopy()), null);
+        var indexRequestDoc = workload.createIndex(options.index.indexSettings.deepCopy());
+        log.atInfo().setMessage("Creating index {} with {}").addArgument(indexName).addArgument(indexRequestDoc).log();
+        client.createIndex(indexName, indexRequestDoc, null);
 
         var docIdCounter = new AtomicInteger(0);
         var allDocs = workload.createDocs(options.totalDocs)
-            .map(doc -> new DocumentReindexer.BulkDocSection(indexName + "_ " + docIdCounter.incrementAndGet(), doc.toString()))
+            .map(doc -> {
```
Comment (gregschohn) [inline DataGenerator/src/main/java/org/opensearch/migrations/data/WorkloadGenerator.java:54]: I would have liked to have used peek() here, but I didn't want to have to lose the sonarqube check for the whole file.

File: DataGenerator/src/main/java/org/opensearch/migrations/data/WorkloadGenerator.java [hunk 3]
```diff
@@ -43,11 +45,25 @@ public void generate(WorkloadOptions options) {
...
+                    .addArgument(indexName)
+                    .addArgument(() -> {
+                            try {
+                                return new ObjectMapper().writeValueAsString(n);
+                            } catch (JsonProcessingException e) {
+                                throw new RuntimeException(e);
+                            }
+                        }
+                    ).log();
+                return new DocumentReindexer.BulkDocSection(indexName + "_" + docIdCounter.incrementAndGet(), doc.toString());
```
Comment (gregschohn) [inline DataGenerator/src/main/java/org/opensearch/migrations/data/WorkloadGenerator.java:65]: notice that "_ " has been replaced with just "_"

**Review by chelma — APPROVED** [review]
Easier to read now - thanks Greg!

File: DataGenerator/src/main/java/org/opensearch/migrations/data/workloads/Workloads.java [hunk 4]
```diff
@@ -7,10 +7,10 @@
...
 @AllArgsConstructor
 public enum Workloads {
-    Geonames(Geonames::new),
-    HttpLogs(HttpLogs::new),
-    Nested(Nested::new),
-    NycTaxis(NycTaxis::new);
+    GEONAMES(Geonames::new),
+    HTTP_LOGS(HttpLogs::new),
+    NESTED(Nested::new),
+    NYC_TAXIS(NycTaxis::new);
```
[the remark this answers is not in the document]
Comment (gregschohn) [reply DataGenerator/src/main/java/org/opensearch/migrations/data/workloads/Workloads.java:13]: thx - done

**Pull request merged**